io/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -280,7 +279,7 @@ private object RTS {
       // and result in zero heap allocations for the happy path.
       if (caught && stack.isEmpty) stack.push(IdentityCont)
 
-      finalizer
+      (errorHandler, finalizer)
```

Comment:
The "zero heap allocations" for the happy path comment is no longer true. 😢 

The tuple is always allocated.